typescript/packages/lattice-client-react/src/context/lattice-client-provider.tsx

```diff
@@ -14,10 +14,12 @@ export function LatticeClientProvider({
   React.useEffect(() => {
     if (!client) return;
 
-    void client.instance.connect();
+    if (client.connection.status !== 'initial') return;
```

Comment:
Maybe this status could be named better, or a little explanation of *why* we don't want to attempt when it's in the `initial` state might be nice. 

So if the states are:

```
 | 'initial'
  | 'connected'
  | 'pending'
  | 'error'
  | 'disconnected';
```
What about the attempting (re)connections if we're in the `disconnected` or `error`? Should this effect be first connection only?

Reply:
This is the other way around, if it's _not_ in the initial state, don't try to connect. I've added a comment though 👍🏻